CMakeLists.txt

```diff
@@ -1,19 +1,51 @@
 cmake_minimum_required(VERSION 3.1)
 
-project(dbg_macro)
+project(dbg_macro LANGUAGES CXX)
 
-set(CMAKE_CXX_STANDARD 11)
+###############################################################################
+# CMake Options
+###############################################################################
+
+option(DBG_MACRO_BUILD_EXAMPLES OFF)
+option(DBG_MACRO_BUILD_TESTS OFF)
+
+###############################################################################
+# Add dbg_macro Library
+###############################################################################
 
 add_library(${PROJECT_NAME} INTERFACE)
-target_include_directories(${PROJECT_NAME} INTERFACE ".")
+add_library(${PROJECT_NAME}::${PROJECT_NAME} ALIAS ${PROJECT_NAME})
+target_include_directories(${PROJECT_NAME} INTERFACE
+	$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+	$<INSTALL_INTERFACE:include>)
+
+
+###############################################################################
+# Build Examples
+###############################################################################
 
-add_executable(${PROJECT_NAME}-example tests/example.cpp)
-target_compile_options(${PROJECT_NAME}-example PRIVATE -Wall -Wextra -pedantic)
-target_link_libraries(${PROJECT_NAME}-example ${PROJECT_NAME})
+if(DBG_MACRO_BUILD_EXAMPLES)
+	macro(dbg_macro_add_example EXAMPLENAME)
+		add_executable(${EXAMPLENAME} ${ARGN})
+		target_compile_features(${EXAMPLENAME} PRIVATE cxx_std_11)
+		target_link_libraries(${EXAMPLENAME} PRIVATE dbg_macro::dbg_macro)
+		set_target_properties(${EXAMPLENAME} PROPERTIES FOLDER examples)
+	endmacro()
+	add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/examples")
+endif()
 
-add_executable(${PROJECT_NAME}-tests tests/tests.cpp)
-target_compile_options(${PROJECT_NAME}-tests PRIVATE -Wall -Wextra -pedantic)
-target_link_libraries(${PROJECT_NAME}-tests ${PROJECT_NAME})
+###############################################################################
+# Build Tests
+###############################################################################
 
-enable_testing()
-add_test(${PROJECT_NAME} ${PROJECT_NAME}-tests)
+if(DBG_MACRO_BUILD_TESTS)
+	enable_testing()
+	macro(dbg_macro_add_test TESTNAME)
+		add_executable(${TESTNAME} ${ARGN})
+		target_compile_features(${TESTNAME} PRIVATE cxx_std_11)
+		target_link_libraries(${TESTNAME} PRIVATE dbg_macro::dbg_macro)
+		add_test(NAME ${TESTNAME} COMMAND ${TESTNAME})
+		set_target_properties(${TESTNAME} PROPERTIES FOLDER tests)
+	endmacro()
+	add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/tests")
+endif()
```

Comment:
Install the target
```suggestion
endif()

###############################################################################
# Install
###############################################################################

install(TARGETS dbg_macro EXPORT dbg_macroConfig)
install(EXPORT dbg_macroConfig NAMESPACE dbg_macro:: DESTINATION lib/cmake/dbg_macro)
install(DIRECTORY include DESTINATION .)
```

Reply:
Done! I am so trash with CMake Configurations, really! Do you have any good guide on this stuff?

Reply:
Oh I barely know what I'm doing.

https://cliutils.gitlab.io/modern-cmake/ is a good source and lists other good ones too.